src/lib/authenticate-cloud.ts

```diff
@@ -0,0 +1,39 @@
+import axios, { AxiosPromise, AxiosRequestConfig, AxiosResponse } from 'axios';
+
+interface AuthenticateCloudParams {
+  AM_URL: string;
+  realm: string;
+  username: string;
+  password: string;
+}
+interface CloudAuth {
+  tokenId: string;
+  successUrl: string;
+  realm: string;
+}
+async function authenticateCloud({
+  realm,
+  AM_URL,
+  username,
+  password,
+}: AuthenticateCloudParams): Promise<AxiosResponse<CloudAuth>> {
+  try {
+    const config: AxiosRequestConfig = {
+      baseURL: AM_URL.slice(-2),
+      method: 'POST',
+      url: `/openam/json/realms/root/realms/${realm}/authenticate`,
```

Comment:
You'll likely want to construct this in a more flexible manner in the future as the realms can get pretty hairy in reality.

Reply:
I think you should authenticate to top-level realm - i.e. this endpoint: `{{AM_URL}}/json/authenticate`